File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        
        def dfs(i, subset):
            if i >= len(nums):
                if subset[:] not in res:
                    res.append(subset[:])
                return
            
            subset.append(nums[i])
            dfs(i + 1, subset)

            subset.pop()
            dfs(i + 1, subset)



        dfs(0, [])

        return res

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int pointer1 = nums[i];
            for (int j = i+1; j < nums.length; j++) {
                if (pointer1 == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
}
